python/kserve/kserve/model.py

```diff
@@ -75,19 +89,43 @@ def __init__(self, name: str):
         self._http_client_instance = None
         self._grpc_client_stub = None
 
-    async def __call__(self, body, model_type: ModelType = ModelType.PREDICTOR):
-        request = await self.preprocess(body) if inspect.iscoroutinefunction(self.preprocess) \
-            else self.preprocess(body)
+    async def __call__(self, body, model_type: ModelType = ModelType.PREDICTOR, headers={}):
+        request_id = headers.get("X-RequestID")
```

Comment:
according to the logging format `"X-RequestID": "{{index .Request.Header "X-Request-Id"}}"`, the header name is actually `X-Request-Id`